cpp/src/parquet/properties.h

```diff
@@ -199,7 +199,8 @@ class PARQUET_EXPORT WriterProperties {
           pagesize_(kDefaultDataPageSize),
           version_(ParquetVersion::PARQUET_2_4),
           data_page_version_(ParquetDataPageVersion::V1),
-          created_by_(DEFAULT_CREATED_BY) {}
+          created_by_(DEFAULT_CREATED_BY),
+          integer_annotate_decimal_(false) {}
```

Comment:
Another reason to keep the default false, is to not change enable code paths that weren't used before when people update there library version.

Reply:
I understand the desire for stability, but would like to have a pathway for improved defaults to be released. It's unfortunate we always do a major release while still trying to respect stability, as there's no clear good time to make breaking changes like this.

Reply:
I agree this is a hard problem, and don't have any guidance.  The one thing I would say is that if Arrow is intended to be developer facing, then IMO developers looking to upgrade versions we should be following the "least surprise" principle as much as possible.  Other opinions are valid here.

Reply:
Hi,
Some belated comments:
* Maybe it's because I am not a native English speaker, but I have a hard time understanding what "integer annotate decimal" means. Is "annotate" a word here?
* I agree that the defaults should remain false for the time being.
* While enabling this does allow for a more compact representation, compression might at least partially alleviate that (though unfortunately FIXED_LEN_BYTE_ARRAY doesn't seem to support any of the delta encodings)

Reply:
Filed https://issues.apache.org/jira/browse/PARQUET-2231 because parquet-mr allows DELTA_BYTE_ARRAY on FIXED_LEN_BYTE_ARRAY, while the spec doesn't say so...